### PR TITLE
Bump version v0.23.3

### PR DESCRIPTION
**Pull request by karan6181**
# What does this PR do?
- Bump version to v0.23.3

### CODE REVIEW

File: composer/_version.py [hunk 1]
```diff
@@ -3,4 +3,4 @@
 
 """The Composer Version."""
 
-__version__ = '0.24.0.dev0'
+__version__ = '0.23.3'
```
Comment (snarayan21) [inline composer/_version.py:6]: isn't this supposed to still be 0.24.dev0, and then only on the release branch you change it to 0.23.0? pretty sure that's what's in the doc

**Review by snarayan21 — COMMENTED** [review]
need to verify what the version should be on the PR merged to dev

File: composer/_version.py (hunk 1 above)
Comment (bigning) [reply composer/_version.py:6]: @mvpatel2000 updated the doc to fix the release note auto-generation. Now it needs this change and an extra PR to restore the version to dev

**Comment by karan6181** [comment]
Closing this PR against https://github.com/mosaicml/composer/pull/3414 since the PR source branch must be from the main repo and not from the fork; otherwise, docker credentials won't work.

**Pull request closed without merging**